docs/proposals/backup/backup.md

```diff
@@ -0,0 +1,735 @@
+	// Each resource name has format "namespace/objectname".  For cluster resources, simply use "objectname".
+	// For example, if you have a specific order for pods, such as "pod1, pod2, pod3" with all belonging to the "ns1" namespace,
+	// and a specific order for persistentvolumes, such as "pv4, pv8", you can use the orderedResources field in YAML format as shown below:
+	// ```yaml
+	// orderedResources:
+	//  pods: "ns1/pod1, ns1/pod2, ns1/pod3"
+	//  persistentvolumes: "pv4, pv8"
+	// ```
+	// +optional
+	// +nullable
+	OrderedResources map[string]string `json:"orderedResources,omitempty"`
+
+	// ItemOperationTimeout specifies the time used to wait for asynchronous BackupItemAction operations.
+	// The default value is 1 hour.
+	// +optional
+	ItemOperationTimeout metav1.Duration `json:"itemOperationTimeout,omitempty"`
+}
+
+type BackupStatus struct {
+	// Conditions represent the current state of the backup operation.
+	// +optional
+	Conditions capiv1.Conditions `json:"conditions,omitempty"`
+
+	// Phase represents the current phase of the backup operation.
+	// +optional
+	Phase string `json:"phase,omitempty"`
+
+	// Details provides a detailed status for each backup in each cluster.
+	// +optional
+	Details []*BackupDetails `json:"backupDetails,omitempty"`
+}
+
+type BackupDetails struct {
+	// ClusterName is the Name of the cluster where the backup is being performed.
+	// +optional
+	ClusterName string `json:"clusterName,omitempty"`
+
+	// ClusterKind is the kind of ClusterName recorded in Kurator.
+	// +optional
+	ClusterKind string `json:"clusterKind,omitempty"`
+
+	// BackupNameInCluster is the name of the backup being performed within this cluster.
+	// This BackupNameInCluster is unique in Storage.
+	// +optional
+	BackupNameInCluster string `json:"backupNameInCluster,omitempty"`
+
+	// BackupStatusInCluster is the current status of the backup performed within this cluster.
+	// +optional
+	BackupStatusInCluster *velerov1.BackupStatus `json:"backupStatusInCluster,omitempty"`
+}
+
+// BackupList contains a list of Backup.
+// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type BackupList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Backup `json:"items"`
+}
+```
+
+Here is the details about `Destination` and `ResourceFilter`
+
+```console
+
+// Destination defines a fleet or specific clusters.
+type Destination struct {
+	// Fleet is the name of fleet.
+	// The field, in combination with ClusterSelector, can determine a set of clusters.
+	// In addition to this approach, users can also directly specify clusters through the field Clusters.
+	// +optional
+	Fleet string `json:"fleet,omitempty"`
+
+	// ClusterSelector specifies the selectors to select the clusters within the fleet.
+	// If unspecified, all clusters in the fleet will be selected.
+	// The field will only take effect when Fleet is set.
```

Comment:
is this consistent with policy?

Reply:
Checked it is similar

Reply:
this is the LabelSelector in ResourceFilter within policy. it come from "k8s.io/apimachinery/pkg/apis/meta/v1"
```
// A label selector is a label query over a set of resources. The result of matchLabels and
// matchExpressions are ANDed. An empty label selector matches all objects. A null
// label selector matches no objects.
// +structType=atomic
type LabelSelector struct {
	// matchLabels is a map of {key,value} pairs. A single {key,value} in the matchLabels
	// map is equivalent to an element of matchExpressions, whose key field is "key", the
	// operator is "In", and the values array contains only "value". The requirements are ANDed.
	// +optional
	MatchLabels map[string]string `json:"matchLabels,omitempty" protobuf:"bytes,1,rep,name=matchLabels"`
	// matchExpressions is a list of label selector requirements. The requirements are ANDed.
	// +optional
	MatchExpressions []LabelSelectorRequirement `json:"matchExpressions,omitempty" protobuf:"bytes,2,rep,name=matchExpressions"`
}
```
the diff is we less 	`MatchExpressions []LabelSelectorRequirement `. 
that means, you can not only list matchLabels, but also can use `in` or `NotIn` expression about matchLabels when you set MatchExpressions.
here is the details:
```
// A label selector requirement is a selector that contains values, a key, and an operator that
// relates the key and values.
type LabelSelectorRequirement struct {
	// key is the label key that the selector applies to.
	// +patchMergeKey=key
	// +patchStrategy=merge
	Key string `json:"key" patchStrategy:"merge" patchMergeKey:"key" protobuf:"bytes,1,opt,name=key"`
	// operator represents a key's relationship to a set of values.
	// Valid operators are In, NotIn, Exists and DoesNotExist.
	Operator LabelSelectorOperator `json:"operator" protobuf:"bytes,2,opt,name=operator,casttype=LabelSelectorOperator"`
	// values is an array of string values. If the operator is In or NotIn,
	// the values array must be non-empty. If the operator is Exists or DoesNotExist,
	// the values array must be empty. This array is replaced during a strategic
	// merge patch.
	// +optional
	Values []string `json:"values,omitempty" protobuf:"bytes,3,rep,name=values"`
}
```